content/docs/tech-insights/track-react-version/index.md

```diff
@@ -0,0 +1,153 @@
+
+1. Visit Tech Insights, select the **Data Sources** tab, and press **ADD DATA SOURCE**.
+
+   ![a button on a web interface](./Add_Data_Source.png)
+
+2. Enter a descriptive **Name** and **Description**.
+
+   ![About section input fields](./Data_About_Section.png)
+
+3. Set the Data Provider **Type** to `Component Repository File`, **Location** to `package.json`. Select a value for **Select Entity to test data source against** and press **TEST**.
+
+    ![Data Provider section input fields](./Data_Provider_Section.png)
+
+4. Test results are displayed.
+
+    ![Data Provider test results displayed](./Data_Test_Results_Section.png)
+
+5. Create multiple Facts:
+    1. Set **Parser** to `JSON with JSONata syntax`.
+    2. Set the first Fact to:
+        | Field Name | Value |
+        | --- | --- |
+        | Fact Name | React version |
+        | JSONata query | $exists(resolutions."@types/react") ? resolutions."@types/react" : $exists(dependencies.react) ? dependencies.react : "" |
+        | Type | String |
+
+    3. Press **ADD FACT** and set the first Fact to:
+        | Field Name | Value |
+        | --- | --- |
+        | Fact Name | Has React dependency |
+        | JSONata query | $exists(resolutions."@types/react") ? $boolean(resolutions."@types/react") : $exists(dependencies.react) ? $boolean(dependencies.react) : false |
+        | Type | Boolean | 
+
+    ![Data Facts section input fields](./Data_Facts_Section.png)
+
+6. Press **CHECK FACTS**.
+
+    ![Data Facts results displayed](./Data_Facts_Results_Section.png)
+
+7. Use the **Applies to** filter to target this data source at some components which you expect to have React versions. We recommend starting with a highly targeted filter for initial experimentation and iteration. You can widen the filter later to capture more results. 
+
+    ![About Data Source section displayed](./Applies_To_Section.png)
+
+8. Press **SAVE**.
+
+## Visualise the distribution of React versions being used
+Data Source comes with a built-in visualisation panel which lets us get an overview of the Dockerfile base image versions being used. When viewing a Data Source, expand the “Facts visualization” section to see it.
+
+![Data Visualisation results displayed](./Data_Visualization_Section.png)
+
+This chart tells us:
+
+1. 17% of the Components that this Data Source targets are using version `17.0.59`.
+2. 17% are using version `15.4.0`.
+3. 67% of Components do not have a React version at all. This is the **Empty Results** section.
+
+In the next section, we will create a Check which can show a pass or fail result to app dev teams to tell them if they need to upgrade their React version.
+
+## Create a check that shows which software is not using a supported React version
+
+We have a data source telling us what React version is present in our repositories, but is also including non-React components. Let's omit the non-React components from the results by creating a check.
```

Comment:
```suggestion
We have a data source telling us what React version is present in our repositories, but is also including non-React components. Let's omit the non-React components from the results by creating a check.
```

The last sentence somehow feels funky in this context, I don't know how to modify though

Reply:
Thank you, I will have a think on this. I stole it from the Dockerfile tutorial.